backend/index.js

```diff
@@ -0,0 +1,126 @@
+import * as THREE from 'three'
+import * as Shaders from './shaders'
+import saveAs from 'file-saver'
+
+// controls: wasd + rf + qe + mouse
+require('three-fly-controls')(THREE)
+
+var container, camera, scene, renderer, controls
+
+const MAX_IMG_PX = [512, 512] // Maximum size in pixels the heightmap can be
+
+function getHeightMap(img) { // Extracts heightmap data
+	if (img.width > MAX_IMG_PX[0] || img.height > MAX_IMG_PX[1]) {
+		return `Image dimensions are too large. Max dimensions are ${MAX_IMG_PX[0]} by ${MAX_IMG_PX[1]}`
+	}
+	let canvas = document.createElement('canvas')
+	canvas.width = img.width
+	canvas.height = img.height
+	let context = canvas.getContext('2d')
+	context.drawImage(img, 0, 0)
+	let data = context.getImageData(0, 0, img.width, img.height).data
+	return data
+}
+
+/* Main function that creates everything
+ * hmap - heightmap data
+ * hmap_width, hmap_height - width and height of heightmap in pixels
+ * scale_plane - multiplier to scale generated mesh by
+ * scale_height - multiplier to scale height values by
+ */
+function init(hmap, hmap_width, hmap_height, scale_plane=0.01, scale_height=1) {
+	container = document.getElementById('container')
+	camera = new THREE.PerspectiveCamera(20, window.innerWidth / window.innerHeight, 1, 10000)
+	camera.position.z = 1800
+	scene = new THREE.Scene()
+	scene.background = new THREE.Color(0xffffff)
+
+	let plane = [hmap_width * scale_plane, hmap_height * scale_plane, hmap_width - 1, hmap_height - 1]
+	let geom = new THREE.PlaneBufferGeometry(...plane)
```

Comment:
Since `plane` doesn't seem to be used elsewhere, the arguments could be inside the geom call instead.
```js
let geom = new THREE.PlaneBufferGeometry(hmap_width * scale_plane, hmap_height * scale_plane, hmap_width - 1, hmap_height - 1)
```